### DISCUSSION TITLE
Using Atlas Search in Morphia

### DISCUSSION THREAD

**Discussion by DDani** [MorphiaOrg/morphia#2000] [Q&A]
Hi, 
  currently, we have our Mongo servers running at Atlas and as a feature request, we must implement full-text search in our service, which uses Morphia, and this feature will affect one of our mapped classes. Our project uses yet Morphia 1.6.1.
  
Atlas provides a command that is not natively supported by Mongo, but in our case simplifies a lot our use-case (we don't have, for example, to index all our documents on Elastic Search).

By looking at the AggregationPipeline interface of Morphia 1.61 or the Javadocs of the Aggregation interface in the last version (2.3-SNAPSHOT), we don't see how to achieve this, and we were wondering if there is some way, like pass a native object to one of the stages of the pipeline.

There is any workaround for this?.

Thanks in advance.

**Comment by evanchooly** [comment]
I haven't really looked at Atlas search just yet. You're the first to ask about it.   It's been on the periphery of things to tackle but other issues have been more pressing. I can take a look at it and see what support in Morphia might look like.

**Reply by DDani** [reply]
Thanks, I would really appreciate it. Of course, anything I can do to help, I’ll will do it.
